frontend/src/metabase/core/components/AccordionList/AccordionListCell.jsx

```diff
@@ -137,8 +139,9 @@ export const AccordionListCell = ({
       >
         <span
           className={cx(
-            "p1 flex-auto flex align-center",
+            "p1 flex align-center",
             isClickable ? "cursor-pointer" : "cursor-default",
+            (!extra || itemExtraIsRightAligned) && "flex-auto",
```

Comment:
We need `flex-auto` in the case where the `extra`, the component returned by `renderItemExtra`, should be aligned to the right of the flex box, or when there is no `extra`. In these cases the span needs to grow to fill the rest of the container. To make the `extra` component left aligned, we omit this.
Note we can't use `justify-content: space-between` on the parent container because the clickable area of the span matters - it needs to expand to fill the rest of the container when `itemExtraIsRightAligned=true`.

Reply:
There's a way, a bit hacky but no API change required, to not have to deal with `flex-auto`.
1. revert the change regarding `flex-auto` and `itemExtraIsRightAligned` back.
2. Use `Flex` with a lot of `flex-grow` to force the ratio of the extra item to grow more than the item itself e.g. I use this in `frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx`
    ```ts
    <Flex
      style={{
        flexGrow: 9999,
      }}
    >
      <PickerSpinner size={16} borderWidth={2} />
    </Flex>
    ```

    it will look as if the item, Orders, in this case hasn't grown at all, but in reality it's grown 1 to 9999 which is negligible
    ![image](https://github.com/metabase/metabase/assets/1937582/c4313da5-7110-4dcd-afb4-25bd3e79dded)

#### Pros
- No API change is needed. 
#### Cons
- A bit hacky? But this utilizes flexbox behavior.

I want to hear @metabase/core-frontend-42 opinion on this. But I'd favor this approach over introducing API changes (`itemExtraIsRightAligned` and `(!extra || itemExtraIsRightAligned)`) in this case since it's easier to maintain because we don't need to deal with the new prop which could potentially increase surface area of bugs.

Reply:
Per [this thread](https://metaboat.slack.com/archives/C505ZNNH4/p1687943176434779) with the FE engineers I've updated this code now. Instead of using `extra` to render the spinner I'm now putting the spinner inside the clickable span on the left
<img width="354" alt="image" src="https://github.com/metabase/metabase/assets/39073188/4b3d35fe-886e-4dce-a7dd-952c1c29a908">

This needs a new API change with `showSpinner` but it's much tidier without CSS hacks. We can also reuse `showSpinner` for rendering the spinner in the case of initially syncing databases, so we don't need the `renderSectionExtra` prop anymore:
![image](https://github.com/metabase/metabase/assets/39073188/58d60c10-fa30-424b-893c-3670da7b12f0)